worker-workflow/src/main/java/com/github/cafdataprocessing/workflow/ArgumentsManager.java

```diff
@@ -208,12 +274,126 @@ else if (type.equals("cd")){
     @SuppressWarnings("unused")
     public void checkHealth() {
         try {
-            final Setting setting = settingsApi.getSetting("healthcheck");
+            settingsApi.getSetting("healthcheck");
         }
         catch (final ApiException ex){
             if(ex.getCode()!=404){
                 throw new RuntimeException(ex.getMessage(), ex);
             }
         }
     }
+
+    private static boolean isCacheableResponse(final Response response)
+    {
+        return !response.request().urlString().contains("healthcheck") && response.isSuccessful();
+    }
+
+    private boolean shouldForceCacheRefresh(
+        final String settingName,
+        final List<String> scopes,
+        final List<String> priorities,
+        final Optional<Long> settingsServiceLastUpdateTimeMillisOpt)
+    {
+        if (!settingsServiceLastUpdateTimeMillisOpt.isPresent()) {
+            return false;
+        }
+        final SettingsServiceLastAccessTimeMapKey key = SettingsServiceLastAccessTimeMapKey.from(settingName, scopes, priorities);
+        final Long settingsServiceLastAccessTimeMillis = settingsServiceLastAccessTimeMap.get(key);
+        if (settingsServiceLastAccessTimeMillis == null) {
+            return true;
+        }
+        if (settingsServiceLastUpdateTimeMillisOpt.get() > settingsServiceLastAccessTimeMillis) {
+            LOG.debug(String.format("Forcing cache refresh for: %s because the last update time: %s is greater than "
+                + "the last access time: %s", key, settingsServiceLastUpdateTimeMillisOpt.get(), settingsServiceLastAccessTimeMillis));
+            return true;
+        }  else {
+            LOG.debug(String.format("NOT forcing cache refresh for: %s because the last update time: %s is NOT greater "
+                + "than the last access time: %s", key, settingsServiceLastUpdateTimeMillisOpt.get(),
+                settingsServiceLastAccessTimeMillis));
+            return false;
+        }
+    }
+
+    private ResolvedSetting forceCacheRefresh(final SettingsServiceApiCall settingsServiceApiCall) throws ApiException
+    {
+        try {
+            okHttpClient.interceptors().add(forceCacheRefreshInterceptor);
```

Comment:
I'm not sure I understand what's going on here.  Won't this impact more than just the call being refreshed?  I.e. isn't `okHttpClient` shared across threads?

Reply:
Yes actually it would, thanks. I've reworked this to use 2 clients with different interceptors instead of adding and removing interceptors on a single client: https://github.com/CAFDataProcessing/worker-workflow/pull/88/commits/aa6fefce483b07010dfbb8aa1f5c35c4d47c1383